### PR TITLE
converted current debugging statements in LLVM codegen to jit-logging statements #48771

### PR DESCRIPTION
**Pull request by huiguoo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #49121 added macros in jit logging to check whether loggings are enabled; replaced similar checks in LLVM codegen with such macros
* **#49040 converted current debugging statements in LLVM codegen to jit-logging statements #48771**

Differential Revision: [D25407356](https://our.internmc.facebook.com/intern/diff/D25407356)

### CODE REVIEW

File: torch/csrc/jit/tensorexpr/llvm_codegen.cpp [hunk 1]
```diff
@@ -513,34 +512,33 @@ void LLVMCodeGenImpl::emitKernel(
...
 
-#if DEBUG_PRINT
-  llvm::errs() << *module_;
-  llvm::SmallVector<char, 0> asmBuffer;
-  llvm::raw_svector_ostream asmStream(asmBuffer);
-  llvm::legacy::PassManager PM;
-  TM_->addPassesToEmitFile(
+  // print graph debug info after optimization
+  asmBuffer.set_size(0);
+  if (is_enabled(__FILE__, ::torch::jit::JitLoggingLevels::GRAPH_DEBUG)) {
```
Comment (ZolotukhinM) [inline torch/csrc/jit/tensorexpr/llvm_codegen.cpp:531]: Nit: might be a good idea to wrap it with a macro and make it commonly available (if you decide to do that, please put that into a separate PR).

File: torch/csrc/jit/tensorexpr/llvm_codegen.cpp [hunk 2]
```diff
@@ -513,34 +512,33 @@ void LLVMCodeGenImpl::emitKernel(
...
-  llvm::errs() << *module_;
-#endif
   if (llvm::verifyFunction(*fn_, &llvm::outs())) {
     throw std::runtime_error("Function verification failed");
   }
 
-  // print graph debug info.
-  std::string fnstr;
-  llvm::raw_string_ostream FS(fnstr);
-  fn_->print(FS);
```
Comment (ZolotukhinM) [inline torch/csrc/jit/tensorexpr/llvm_codegen.cpp:526]: This print has disappeared, is it duplicating what's printed in the module?

File: torch/csrc/jit/tensorexpr/llvm_codegen.cpp [hunk 3]
```diff
@@ -513,34 +512,33 @@ void LLVMCodeGenImpl::emitKernel(
...
-  GRAPH_DEBUG("LLVM Function:\n", FS.str(), "\n");
+  // print graph debug info before optimization
+  llvm::SmallVector<char, 0> asmBuffer;
+  llvm::raw_svector_ostream asmStream(asmBuffer);
+  if (is_enabled(__FILE__, ::torch::jit::JitLoggingLevels::GRAPH_DEBUG)) {
+    module_->print(asmStream, nullptr);
+  }
+  GRAPH_DEBUG(asmStream.str().str(), "\n");
```
Comment (ZolotukhinM) [inline torch/csrc/jit/tensorexpr/llvm_codegen.cpp:525]: Nit: maybe add a header "LLVM module before optimizations" or something like that?

**Review by ZolotukhinM — APPROVED** [review]
Awesome!

File: torch/csrc/jit/tensorexpr/llvm_codegen.cpp (hunk 2 above)
Comment (huiguoo) [reply torch/csrc/jit/tensorexpr/llvm_codegen.cpp:526]: Yes, it's a duplicate. `module_->print()` prints the graph info generated by `fn_->print()` plus info associated with the module; it should be more helpful.

File: torch/csrc/jit/tensorexpr/llvm_codegen.cpp (hunk 1 above)
Comment (huiguoo) [reply torch/csrc/jit/tensorexpr/llvm_codegen.cpp:531]: Sounds good. Will go ahead and add it.

Comment (huiguoo) [reply torch/csrc/jit/tensorexpr/llvm_codegen.cpp:531]: To confirm, are you suggesting adding a macro in `jit/jit_log.h` and making it available to all code that includes this header file? Using this macro, we can disable some computation required for loggings when logging is not enabled.

Comment (ZolotukhinM) [reply torch/csrc/jit/tensorexpr/llvm_codegen.cpp:531]: Yes, exactly that.

Comment (huiguoo) [reply torch/csrc/jit/tensorexpr/llvm_codegen.cpp:532]: Addressed in PR #49121.
